File: Backend/MuseumAPI/Models/Artist.cs
namespace MuseumAPI.Models
{
    public class Artist
    {
        public long Id { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public DateTime? BirthDate { get; set; }
        public string? BirthPlace { get; set; }
        public string? Education { get; set; }
        public string? Movement { get; set; }

        // Hidden from the API because it's not in the DTO
        public virtual ICollection<Painting> Paintings { get; set; } = null!;
        public virtual ICollection<Museum> Museums { get; set; } = null!;
        public virtual ICollection<Exhibition> Exhibitions { get; set; } = null!;

        public virtual long? UserId { get; set; }
        public virtual User? User { get; set; } = null!;
    }
}

File: Frontend/src/models/Artist.ts
import { Exhibition } from "./Exhibition";
import { Museum } from "./Museum";
import { Painting } from "./Painting";
import { User } from "./User";

export interface Artist {
    id?: number;
    firstName: string;
    lastName: string;
    birthDate: Date;
    birthPlace: string;
    education: string;
    movement: string;
    paintings?: Painting[];
    museums?: Museum[];
    exhibitions?: Exhibition[];

    userId?: number;
    user?: User;

    [key: string]: any;
}
